net-misc/seafile-client/seafile-client-6.0.4.ebuild

```diff
@@ -0,0 +1,51 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+inherit cmake-utils
+
+DESCRIPTION="Seafile desktop client"
+HOMEPAGE="https://github.com/haiwen/seafile-client/ http://www.seafile.com/"
+SRC_URI="https://github.com/haiwen/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="shibboleth"
+
+RDEPEND="net-libs/libsearpc
+	=net-libs/ccnet-${PV}
+	=net-misc/seafile-${PV}
+	>=dev-libs/libevent-2.0
+	>=dev-libs/jansson-2.0
+	dev-libs/openssl:0
+	dev-db/sqlite:3
+
+	dev-qt/qtcore:5
+	dev-qt/qtgui:5
+	dev-qt/qtwidgets:5
+	dev-qt/qtnetwork:5
+	dev-qt/qtdbus:5
+	shibboleth? ( || ( dev-qt/qtwebkit:5 dev-qt/qtwebengine:5[widgets] ) )"
+DEPEND="${RDEPEND}
+	dev-qt/linguist-tools:5"
+
+CMAKE_MIN_VERSION=2.8.9
+CMAKE_BUILD_TYPE=Release
+
+src_prepare() {
+	if use shibboleth ; then
+		if ! has_version "dev-qt/qtwebengine:5[widgets]" ; then
+			sed -i -e 's/IF(WIN32 OR DETECTED_QT_VERSION VERSION_LESS 5.6.0)/IF(TRUE)/' CMakeLists.txt || die
+		fi
+	fi
+	default
```

Comment:
But in general, this is an okay way to prevent the build system from trying to be smart?